nav2_controller/src/nav2_controller.cpp

```diff
@@ -63,6 +70,15 @@ ControllerServer::on_configure(const rclcpp_lifecycle::State & state)
 {
   RCLCPP_INFO(get_logger(), "Configuring controller interface");
 
+  get_parameter("progress_checker_plugin", progress_checker_id_);
+  if (progress_checker_id_ == default_progress_checker_id_) {
+    declare_parameter(default_progress_checker_id_ + ".plugin", default_progress_checker_type_);
```

Comment:
I think that might be missing.

Reply:
Do we have to mention these? The user does not need to declare them as that will be done in nav2_util::get_plugin_type_param automatically.

Reply:
I think the user might need to set these parameters if they are not using a default plugin. `nav2_util::get_plugin_type_param` just declares (with no default value) and retrieves a parameter, the user will need to set the `<name>.plugin` parameter when using a custom plugin.

Reply:
Ok done